scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowSCollectionFunctions.scala

```diff
@@ -17,167 +17,208 @@
+      .toCompletableFuture
+}
+
 /**
  * Enhanced version of [[com.spotify.scio.values.SCollection SCollection]] with TensorFlow methods.
  */
 class TensorFlowSCollectionFunctions[T: ClassTag](@transient val self: SCollection[T])
-  extends Serializable {
+    extends Serializable {
 
   /**
    * Predict/infer/forward-pass on pre-trained GraphDef.
    *
    * @param graphUri URI of pre-trained/saved TensorFlow model
    * @param fetchOps names of [[org.tensorflow.Operation]]s to fetch the results from
-   * @param config configuration parameters for the session specified as a serialized
-   * `org.tensorflow.framework.ConfigProto` protocol buffer.
-   * @param inFn translates input elements of T to map of input-operation ->
-   * [[org.tensorflow.Tensor Tensor]]. This method takes ownership of the
-   * [[org.tensorflow.Tensor Tensor]]s.
-   * @param outFn translates output of prediction from map of output-operation ->
-   * [[org.tensorflow.Tensor Tensor]], to elements of V. This method takes
-   * ownership of the [[org.tensorflow.Tensor Tensor]]s.
+   * @param config   configuration parameters for the session specified as a serialized
+   *                 `org.tensorflow.framework.ConfigProto` protocol buffer.
+   * @param inFn     translates input elements of T to map of input-operation ->
+   *                 [[org.tensorflow.Tensor Tensor]]. This method takes ownership of the
+   *                 [[org.tensorflow.Tensor Tensor]]s.
+   * @param outFn    translates output of prediction from map of output-operation ->
+   *                 [[org.tensorflow.Tensor Tensor]], to elements of V. This method takes
+   *                 ownership of the [[org.tensorflow.Tensor Tensor]]s.
    */
-  def predict[V: ClassTag, W](graphUri: String,
+  def predict[V: ClassTag, W](graphUri: String, fetchOps: Seq[String], config: Array[Byte] = null)(
+    inFn: T => Map[String, Tensor[_]])(outFn: (T, Map[String, Tensor[_]]) => V): SCollection[V] =
+    self.parDo(new GraphPredictDoFn[T, V](graphUri, fetchOps, config, inFn, outFn))
+
+  /**
+    * Predict/infer/forward-pass on pre-trained SaveModelBundle.
+    *
+    * @param uri URI of pre-trained/saved TensorFlow model
+    * @param fetchOps names of [[org.tensorflow.Operation]]s to fetch the results from
+    * @param options   configuration parameters for the session specified as a
+    *                 `com.spotify.zoltar.tf.TensorFlowModel.Options`.
+    * @param inFn     translates input elements of T to map of input-operation ->
+    *                 [[org.tensorflow.Tensor Tensor]]. This method takes ownership of the
+    *                 [[org.tensorflow.Tensor Tensor]]s.
+    * @param outFn    translates output of prediction from map of output-operation ->
+    *                 [[org.tensorflow.Tensor Tensor]], to elements of V. This method takes
+    *                 ownership of the [[org.tensorflow.Tensor Tensor]]s.
+    */
+  def predict[V: ClassTag, W](uri: String,
```

Comment:
@ravwojdyla not sure about the naming here. Can it be a bit confusing? 

probably we could type this `uri` like ... `SavedBundleModelUri(uri)` wdyt?

Reply:
`uri` -> `dirPath` or `path`